app/frontend/src/components/designer/settings/FormEventStreamSettings.vue

```diff
@@ -0,0 +1,279 @@
+<script setup>
+import moment from 'moment';
+import { storeToRefs } from 'pinia';
+import { useI18n } from 'vue-i18n';
+import { onMounted, ref } from 'vue';
+
+import { useAppStore } from '~/store/app';
+import { useFormStore } from '~/store/form';
+import { useNotificationStore } from '~/store/notification';
+import { encryptionKeyService } from '~/services';
+
+const { t, locale } = useI18n({ useScope: 'global' });
+const appStore = useAppStore();
+const formStore = useFormStore();
+const notificationStore = useNotificationStore();
+const { form, isRTL } = storeToRefs(formStore);
+
+const algorithms = ref([]);
+const techdocsLink =
+  'https://developer.gov.bc.ca/docs/default/component/chefs-techdocs/Capabilities/Integrations/Event-Stream-Service/';
+
+onMounted(async () => {
+  await fetchAlgorithms();
+});
+
+async function fetchAlgorithms() {
+  try {
+    algorithms.value = [];
+    const result = await encryptionKeyService.listEncryptionAlgorithms();
+    algorithms.value = result.data;
+  } catch (e) {
+    notificationStore.addNotification({
+      text: t('trans.formSettings.fetchEncryptionAlgorithmListError'),
+      consoleError: t('trans.formSettings.fetchEncryptionAlgorithmListError', {
+        error: e.message,
+      }),
+    });
+  }
+}
+
+async function generateKey() {
+  let generatedKey;
+  // when we have more algorithms, move this into some kind of service...
+  if (form.value.eventStreamConfig.encryptionKey.algorithm === 'aes-256-gcm') {
+    const data = crypto.getRandomValues(new Uint8Array(16));
+    const hashBuffer = await window.crypto.subtle.digest('SHA-256', data);
+    generatedKey = Array.from(new Uint8Array(hashBuffer))
```

Comment:
SubtleCrypto does do key generation, but not sure if it has an easy way to convert to string.
```suggestion
    const key = await window.crypto.subtle.generateKey(
      {
        name: 'AES-GCM',
        length: 256,
      },
      true,
      ['encrypt', 'decrypt']
    );
    const rawKey = await window.crypto.subtle.exportKey('raw', key);
    generatedKey = Array.from(new Uint8Array(rawKey))
```

Reply:
ok, added that in. works fine.